lib/azure/armrest/storage_account_service.rb

```diff
@@ -203,6 +203,43 @@ def list_private_images(group = configuration.resource_group)
         get_private_images(storage_accounts)
       end
 
+      # Return the storage account for the virtual machine model +vm+.
+      #
+      def get_from_vm(vm)
+        uri = Addressable::URI.parse(vm.properties.storage_profile.os_disk.vhd.uri)
+
+        # The uri looks like https://foo123.blob.core.windows.net/vhds/something123.vhd
+        name = uri.host.split('.').first # storage name, e.g. 'foo123'
+
+        # Look for the storage account in the VM's resource group first. If
+        # it's not found, look through all the storage accounts.
+        begin
+          acct = get(name, vm.resource_group)
+        rescue Azure::Armrest::NotFoundException => err
+          acct = list_all.find { |s| s.name == name }
```

Comment:
This `list_all` requires only one API call, right? It takes a filter as the argument that you can use.

Do you see in real case that the account is in another resource group?